Fleet Operator crashes when checking FederatedNamespace

### What happened?

Here's the resource manifest:

```yaml
apiVersion: fleet.roboscale.io/v1alpha1
kind: Fleet
metadata:
  creationTimestamp: "2023-07-31T08:10:41Z"
  generation: 2
  labels:
    kubefed.io/managed: "true"
    robolaunch.io/cloud-instance: demo-ci-013
    robolaunch.io/cloud-instance-alias: org-demonstration-76924780
    robolaunch.io/organization: org-demonstration
    robolaunch.io/region: eu-central-1
    robolaunch.io/team: org_demonstration
  name: demo-fleet
  resourceVersion: "92966"
  uid: 1721959b-62c1-4bd7-9c93-d4e09f448d3c
spec:
  discoveryServerTemplate:
    hostname: demo-fleet
    reference: {}
    subdomain: org-demonstration
    type: Server
  hybrid: true
  instances:
  - demo-ci-013
  - cloudy-mini-agv
status:
  discoveryServerStatus:
    created: true
    phase: Ready
    reference:
      apiVersion: robot.roboscale.io/v1alpha1
      kind: DiscoveryServer
      name: demo-fleet-discovery
      namespace: demo-fleet
      resourceVersion: "91992"
      uid: 03c90480-51cd-4bcd-b081-908ea0e5cb12
  namespaceStatus:
    federated: true
    ready: true
    resource:
      created: true
      reference: {}
  phase: Ready
```

Here's the log before crash:

```
2023-08-01T08:00:12Z	INFO	Observed a panic in reconciler: interface conversion: interface {} is nil, not map[string]interface {}	{"controller": "fleet", "controllerGroup": "fleet.roboscale.io", "controllerKind": "Fleet", "Fleet": {"name":"demo-fleet"}, "namespace": "", "name": "demo-fleet", "reconcileID": "0c8bb5fd-9512-4e85-adcd-95bc3548cefd"}
panic: interface conversion: interface {} is nil, not map[string]interface {} [recovered]
	panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 358 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.0/pkg/internal/controller/controller.go:119 +0x1fa
panic({0x1684fa0, 0xc00017d350})
	/usr/local/go/src/runtime/panic.go:884 +0x213
github.com/robolaunch/fleet-operator/pkg/controllers/fleet.(*FleetReconciler).reconcileCheckNamespace(0xc00051dc80, {0x1aa45d0, 0xc000742d80}, 0xc0005d2a80)
	/workspace/pkg/controllers/fleet/check.go:59 +0xa2e
github.com/robolaunch/fleet-operator/pkg/controllers/fleet.(*FleetReconciler).reconcileCheckResources(0xc00051dc80?, {0x1aa45d0, 0xc000742d80}, 0xc0005d2a80?)
	/workspace/pkg/controllers/fleet/fleet_controller.go:204 +0x31
github.com/robolaunch/fleet-operator/pkg/controllers/fleet.(*FleetReconciler).Reconcile(0x7fe75bb56108?, {0x1aa45d0, 0xc000742d80}, {{{0x0?, 0x0?}, {0xc00015b460?, 0x40de67?}}})
	/workspace/pkg/controllers/fleet/fleet_controller.go:97 +0x1e8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x1aa45d0?, {0x1aa45d0?, 0xc000742d80?}, {{{0x0?, 0x15fb040?}, {0xc00015b460?, 0x0?}}})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.0/pkg/internal/controller/controller.go:122 +0xc8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0002cf040, {0x1aa4528, 0xc0005a3e50}, {0x16c8c20?, 0xc0001ca0a0?})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.0/pkg/internal/controller/controller.go:323 +0x35f
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0002cf040, {0x1aa4528, 0xc0005a3e50})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.0/pkg/internal/controller/controller.go:274 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.0/pkg/internal/controller/controller.go:235 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.0/pkg/internal/controller/controller.go:231 +0x587
```

[This line](https://github.com/robolaunch/fleet-operator/blob/f960b5f0eac17e0740c2f241ae36541e029be258/pkg/controllers/fleet/check.go#L59) crashes the operator:

```go
remoteObjectPhase := remoteStatus.(map[string]interface{})["remoteStatus"].(map[string]interface{})["phase"].(string)
```

The root cause is that FederatedNamespace's `.status.clusters[].remoteStatus` field can be `nil`, but operator tries to fetch the phase from remote status in every occasion. Here's the FederatedNamespace:

```yaml
apiVersion: types.kubefed.io/v1beta1
kind: FederatedNamespace
metadata:
  name: demo-fleet
  namespace: demo-fleet
  ownerReferences:
  - apiVersion: fleet.roboscale.io/v1alpha1
    blockOwnerDeletion: true
    controller: true
    kind: Fleet
    name: demo-fleet
    uid: 1721959b-62c1-4bd7-9c93-d4e09f448d3c
  resourceVersion: "94684"
  uid: 53833006-e66c-4c0b-8da0-6fbd7293ea10
spec:
  placement:
    clusters:
    - name: demo-ci-013
    - name: cloudy-mini-agv
status:
  clusters:
  - name: demo-ci-013
    remoteStatus:
      phase: Active
  - name: cloudy-mini-agv
    status: ClusterNotReady
  conditions:
  - lastTransitionTime: "2023-07-31T12:52:26Z"
    lastUpdateTime: "2023-07-31T12:52:26Z"
    reason: CheckClusters
    status: "False"
    type: Propagation
  observedGeneration: 2
```

### What did you expect to happen?

I expect operator to ignore clusters that has no remote status of the resource (FederatedNamespace) mentioned. Fix can be something like this:

```go
if rs, ok := remoteStatus.(map[string]interface{})["remoteStatus"]; ok {
	remoteObjectPhase := rs.(map[string]interface{})["phase"].(string)
	if remoteObjectPhase != "Active" {
		instance.Status.NamespaceStatus.Ready = false
		break
	}
}
```

### How can we reproduce it (as minimally and precisely as possible)?

Can be reproduced by shutting down a connected instance (not the federation host) and inspecting fleet operator logs.

### Kubernetes version

All Kubernetes versions

### Container network interface (CNI) and version

_No response_

--- a/pkg/controllers/fleet/check.go
+++ b/pkg/controllers/fleet/check.go
@@ -56,10 +56,12 @@ func (r *FleetReconciler) reconcileCheckNamespace(ctx context.Context, instance 
 				if unstructuredFedNs.Object["status"] != nil {
 					remoteStatusesInterface := unstructuredFedNs.Object["status"].(map[string]interface{})["clusters"].([]interface{})
 					for _, remoteStatus := range remoteStatusesInterface {
-						remoteObjectPhase := remoteStatus.(map[string]interface{})["remoteStatus"].(map[string]interface{})["phase"].(string)
-						if remoteObjectPhase != "Active" {
-							instance.Status.NamespaceStatus.Ready = false
-							break
+						if rs, ok := remoteStatus.(map[string]interface{})["remoteStatus"]; ok {
+							remoteObjectPhase := rs.(map[string]interface{})["phase"].(string)
+							if remoteObjectPhase != "Active" {
+								instance.Status.NamespaceStatus.Ready = false
+								break
+							}
 						}
 					}
 				} else {

